### DISCUSSION TITLE
How can I reimport datastore automatically with a csv url dataset?

### DISCUSSION THREAD

**Discussion by guille-amarilla** [GetDKAN/dkan#4108] [Q&A]
How can I make the datastore to the reimported periodically, like, for example, everyday, or weekly? Because I have csv url dataset, and I want to automatically update in the datastore without having to edit the dataset manually every time. Thanks in advance.

**Comment by janette** [comment]
@guille-amarilla I am going to assume you have a dataset distribution with a remote url, and that when the data is updated, the file name and path do not change. The way to automate a re-import would be to make the modified date value a "trigger" for re-importing the data to the database from the revised file.
1. Go to /admin/dkan/datastore and check the box for “Last Update (modified)” and save. See https://dkan.readthedocs.io/en/latest/user-guide/guide_datastore_settings.html
2. Then you will need to script an API call that runs on a schedule. This authenticated PATCH would update the modified date of your dataset, and that will generate queues to fetch the updated file and import it when cron runs, or you can also run the queues directly with your script using drush commands. See https://dkan.readthedocs.io/en/latest/user-guide/guide_dataset.html#patch--api-1-metastore-schemas-dataset-items-datasetID

**Comment by guille-amarilla** [comment]
Thanks, @janette that helped a lot.
